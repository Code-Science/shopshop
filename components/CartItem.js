import React from 'react';
import PropTypes from 'prop-types';
import styles from '../styles/components/Cart.module.css';
import { useStateValue } from '../store/StateProvider';

function CartItem({ id, price, image, title, quantity }) {
  const [{ basket }, dispatch] = useStateValue();

  const addOneMore = () => {
    // dispatch the item to the data layer
    dispatch({
      type: 'ADD_TO_BASKET',
      item: {
        id,
        title,
        image,
        price,
      },
    });
  };

  const removeFromBasket = () => {
    // remove item from basket
    dispatch({
      type: 'REMOVE_FROM_BASKET',
      id,
    });
  };

  return (
    <li className={styles.cartItem}>
      <div className={styles.details}>
        <div className={styles.imgBox}>
          <img src={image} alt={title} className={styles.img} />
        </div>
        <div className={styles.title}>
          <p>{title}</p>
          <p> color: blue</p>
        </div>
      </div>
      <div className={styles.priceDetails}>
        <div>
          <p className={styles.small}>Price</p>
          <p>${price}</p>
        </div>
        <div>
          <p className={styles.small}>Quantity</p>
          <div className={styles.btns}>
            <button
              type="button"
              className={styles.minus}
              onClick={removeFromBasket}
            >
              -
            </button>
            <p>{quantity}</p>
            <button type="button" className={styles.plus} onClick={addOneMore}>
              +
            </button>
          </div>
        </div>
        <div>
          <p className={styles.small}>Total</p>
          <p>${(quantity * price).toFixed(2)}</p>
        </div>
      </div>
    </li>
  );
}
CartItem.propTypes = {
  id: PropTypes.number,
  price: PropTypes.number,
  image: PropTypes.string,
  title: PropTypes.string,
  quantity: PropTypes.number,
};

export default CartItem;
