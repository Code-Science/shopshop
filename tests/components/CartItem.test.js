import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import CartItem from '../../components/CartItem';

jest.mock('../../store/StateProvider');

configure({ adapter: new Adapter() });

describe('CartItem Component', () => {
  const mockItem = {
    id: 4,
    title: 'shirt',
    price: 80,
    image: 'image1',
    quantity: 1,
  };
  const wrapper = shallow(<CartItem item={mockItem} />);

  it('expect to render CartItem component', () => {
    expect(wrapper.length).toEqual(1);
  });
  it('snapshot test', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
