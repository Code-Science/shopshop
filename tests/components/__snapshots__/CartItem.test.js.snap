// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem Component snapshot test 1`] = `
<li
  className="cartItem"
>
  <div
    className="details"
  >
    <div
      className="imgBox"
    >
      <img
        className="img"
      />
    </div>
    <div
      className="title"
    >
      <p />
      <p>
         color: blue
      </p>
    </div>
  </div>
  <div
    className="priceDetails"
  >
    <div>
      <p
        className="small"
      >
        Price
      </p>
      <p>
        $
      </p>
    </div>
    <div>
      <p
        className="small"
      >
        Quantity
      </p>
      <div
        className="btns"
      >
        <button
          className="minus"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
        <p />
        <button
          className="plus"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </div>
    </div>
    <div>
      <p
        className="small"
      >
        Total
      </p>
      <p>
        $
        NaN
      </p>
    </div>
  </div>
</li>
`;
